src/types/types.ts

```diff
@@ -1624,6 +1624,114 @@ export interface SendPrivateTransactionOptions {
   fast: boolean;
 }
 
+export enum SimulateAssetType {
+  NATIVE = 'NATIVE',
+  ERC20 = 'ERC20',
+  ERC721 = 'ERC721',
+  ERC1155 = 'ERC1155',
+  SPECIAL_NFT = 'SPECIAL_NFT'
+}
+
+export enum SimulateChangeType {
+  APPROVE = 'APPROVE',
+  TRANSFER = 'TRANSFER'
+}
+
+export interface SimulateAssetChange {
+  assetType: SimulateAssetType;
+  changeType: SimulateChangeType;
+  from: string;
+  to: string;
+
+  /* All - NATIVE, ERC20, ERC721, ERC1555, SPECIAL_NFT */
+  rawAmount: string;
+  amount: string;
+  name?: string;
+  symbol: string;
+  decimals: number;
+
+  /* ERC20, ERC721, ERC1555, SPECIAL_NFT */
+  contractAddress?: string;
+
+  /* ERC20  */
+  logo?: string;
+
+  /* ERC721, ERC1555  */
+  tokenId?: string;
+}
+
+/** Response object for the {@link TransactNamespace.simulateAssetChanges} method. */
+export interface SimulateAssetChangesResponse {
+  changes: SimulateAssetChange[];
+  error?: string;
+}
+
+export interface SimulateExecutionOptions {
+  format: 'FLAT' | 'NESTED';
+}
+
+enum DecodingAuthority {
+  ETHERSCAN = 'ETHERSCAN'
+}
+
+enum CallType {
+  CALL = 'CALL',
+  STATICCALL = 'STATICCALL',
+  DELEGATECALL = 'DELEGATECALL'
+}
+
+interface DecodedCallParam {
+  value: string;
+  name: string;
+  type: string;
+}
+interface DecodedCall {
+  methodName: string;
+  inputs: DecodedCallParam[];
+  outputs: DecodedCallParam[];
+  authority: DecodingAuthority;
+}
+
+interface Call {
```

Comment:
is there any overlap with the existing debug namespace return types like `DebugCallTrace` here?